src/ImageSharp/Formats/Webp/Lossy/YuvConversion.cs

```diff
@@ -312,6 +580,175 @@ public static void YuvToBgr(int y, int u, int v, Span<byte> bgr)
+
+            // OR the different channels.
+            Vector128<byte> rg0 = Sse2.Or(r0, g0);
+            Vector128<byte> rg1 = Sse2.Or(r1, g1);
+            Vector128<byte> rg2 = Sse2.Or(r2, g2);
+            Vector128<byte> rg3 = Sse2.Or(r3, g3);
+            Vector128<byte> rg4 = Sse2.Or(r4, g4);
+            Vector128<byte> rg5 = Sse2.Or(r5, g5);
+
+            ref byte outputRef = ref MemoryMarshal.GetReference(rgb);
+            Unsafe.As<byte, Vector128<byte>>(ref outputRef) = Sse2.Or(rg0, b0);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 16)) = Sse2.Or(rg1, b1);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 32)) = Sse2.Or(rg2, b2);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 48)) = Sse2.Or(rg3, b3);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 64)) = Sse2.Or(rg4, b4);
+            Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, 80)) = Sse2.Or(rg5, b5);
+        }
+
+        // Shuffles the input buffer as A0 0 0 A1 0 0 A2
+        private static void ChannelMixing(
+            Vector128<byte> input0,
+            Vector128<byte> input1,
+            Vector128<byte> shuffle0,
+            Vector128<byte> shuffle1,
+            Vector128<byte> shuffle2,
+            out Vector128<byte> output0,
+            out Vector128<byte> output1,
+            out Vector128<byte> output2,
+            out Vector128<byte> output3,
+            out Vector128<byte> output4,
+            out Vector128<byte> output5)
+        {
+            output0 = Ssse3.Shuffle(input0, shuffle0);
+            output1 = Ssse3.Shuffle(input0, shuffle1);
+            output2 = Ssse3.Shuffle(input0, shuffle2);
+            output3 = Ssse3.Shuffle(input1, shuffle0);
+            output4 = Ssse3.Shuffle(input1, shuffle1);
+            output5 = Ssse3.Shuffle(input1, shuffle2);
+        }
+
+        // Convert 32 samples of YUV444 to R/G/B
+        private static void ConvertYuv444ToRgbSse41(Span<byte> y, Span<byte> u, Span<byte> v, out Vector128<short> r, out Vector128<short> g, out Vector128<short> b)
+        {
+            Vector128<byte> y0 = LoadHigh(y);
+            Vector128<byte> u0 = LoadHigh(u);
+            Vector128<byte> v0 = LoadHigh(v);
+
+            Vector128<ushort> y1 = Sse2.MultiplyHigh(y0.AsUInt16(), K19077.AsUInt16());
+
+            Vector128<ushort> r0 = Sse2.MultiplyHigh(v0.AsUInt16(), K26149.AsUInt16());
+            Vector128<ushort> r1 = Sse2.Subtract(y1.AsUInt16(), K14234.AsUInt16());
+            Vector128<ushort> r2 = Sse2.Add(r1, r0);
+
+            Vector128<ushort> g0 = Sse2.MultiplyHigh(u0.AsUInt16(), K6419.AsUInt16());
+            Vector128<ushort> g1 = Sse2.MultiplyHigh(v0.AsUInt16(), K13320.AsUInt16());
+            Vector128<ushort> g2 = Sse2.Add(y1.AsUInt16(), K8708.AsUInt16());
+            Vector128<ushort> g3 = Sse2.Add(g0, g1);
+            Vector128<ushort> g4 = Sse2.Subtract(g2, g3);
```

Comment:
You can try re-grouping the code to do identical operations in a row. It works better with CPU piplines. I would also try to  group the vector loads of `LoadHigh` (after inlining the method).

```suggestion
            Vector128<ushort> y1 = Sse2.MultiplyHigh(y0.AsUInt16(), K19077.AsUInt16());
            Vector128<ushort> r0 = Sse2.MultiplyHigh(v0.AsUInt16(), K26149.AsUInt16());
            Vector128<ushort> g0 = Sse2.MultiplyHigh(u0.AsUInt16(), K6419.AsUInt16());
            Vector128<ushort> g1 = Sse2.MultiplyHigh(v0.AsUInt16(), K13320.AsUInt16());
            
            Vector128<ushort> r1 = Sse2.Subtract(y1.AsUInt16(), K14234.AsUInt16());
            Vector128<ushort> r2 = Sse2.Add(r1, r0);

            Vector128<ushort> g2 = Sse2.Add(y1.AsUInt16(), K8708.AsUInt16());
            Vector128<ushort> g3 = Sse2.Add(g0, g1);
            Vector128<ushort> g4 = Sse2.Subtract(g2, g3);
```